pages/resources/[page].js

```diff
@@ -1,75 +1,254 @@
+    paid: string,
+  }),
+};
+
+Resources.defaultProps = {
+  initialValues: {
+    category: '',
+    q: '',
+    languages: [],
+    paid: '',
+  },
+};
+
+function Resources({ initialValues }) {
   const router = useRouter();
   const { pathname, query } = router;
-  const currentPage = parseInt(query.page, 10);
+  const { page, category, languages, paid, q } = query;
+  const currentPage = parseInt(page, 10);
 
-  if (!isNumber(currentPage)) {
-    // TODO: Handle situation where user tried some funny business, like `/resources/fuck-you`
-    throw new Error(`'currentPage' is not a number`);
+  // eslint-disable-next-line no-restricted-globals
+  if (page && isNaN(page)) {
+    router.push({ pathname: pathname.replace('[page]', '1') });
   }
+  const [errorMessage, setErrorMessage] = useState(null);
 
-  const [isLoading, setIsLoading] = useState(true);
   const [resources, setResources] = useState([]);
   const [totalPages, setTotalPages] = useState(currentPage);
 
+  const [allCategories, setAllCategories] = useState([]);
+  const [allLanguages, setAllLanguages] = useState([]);
+
+  const costOptions = [
+    { label: 'Paid', value: 'true' },
+    { label: 'Free', value: 'false' },
+  ];
+
+  const handleError = error =>
+    error.errors
+      ? error.errors['not-found'].message
+      : 'There was a problem loading those resources.';
+
+  const handleEndpoint = () => {
+    if (q) {
+      return getResourcesBySearch({ page, category, languages, paid, q });
+    }
+    return getResourcesPromise({ page, category, languages, paid });
+  };
+
+  const handleSubmit = (values, actions) => {
+    const emptyQueryParameters = Object.entries(values).filter(
+      item => item[1] === null || !item[1].length,
+    );
+    const activeParameters = omit(
+      values,
+      emptyQueryParameters.map(parameter => parameter[0]),
+    );
+
+    updateQuery(activeParameters);
+    setTimeout(() => {
+      actions.setSubmitting(false);
+    }, 500);
+  };
+
   useEffect(() => {
-    getResourcesPromise({ page: query.page })
+    Promise.all([getResourcesByCategories(), getResourcesByLanguages()])
+      .then(([categoriesResponse, languagesResponse]) => {
+        const {
+          data: { data: categoriesData },
+        } = categoriesResponse;
+        const {
+          data: { data: languagesData },
+        } = languagesResponse;
+
+        setAllLanguages(
+          languagesData.map(languageObject => {
+            return {
+              value: languageObject.name.toLowerCase(),
+              label: languageObject.name,
+            };
+          }),
+        );
+        setAllCategories(
+          categoriesData.map(categoryObject => {
+            return {
+              value: categoryObject.name.toLowerCase(),
+              label: categoryObject.name,
+            };
+          }),
+        );
+      })
+      .catch(error => {
+        setErrorMessage(handleError(error));
+      });
+
+    setErrorMessage(null);
```

Comment:
Wouldn't this clear any error messages even even if an asynchronous one was supposed to exist? I think you mean to call this during submit and on reset

Reply:
Totally. Thanks for catching it